### PR TITLE
remove_kernels.sh

### PR DESCRIPTION
**Pull request by tompods**

### CODE REVIEW

File: remove_kernels.sh [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+       else
+         exit_message "Unknown answer: $answer."
+       fi
+       ;;
+    *) exit_message "Unknown answer: $answer."
+  esac
+}
+
+prepare_systemd_service() {
+  # Once there's a list of the kernel-related packages the user wants to
```
Comment (AlexBaranowski) [inline remove_kernels.sh:109]: Tomek I knew that you will get that service 👍🏼 working ! 😸

**Review by AlexBaranowski — CHANGES_REQUESTED** [review]
Obligatory:

- Comment about mapfile
- Add author (you could use alias if you want)

Maybe:
- use /etc/grub* files instead of the hardcoded paths.

File: remove_kernels.sh [hunk 2]
```diff
@@ -0,0 +1,143 @@
+#!/bin/bash 
+# description: remove kernels and GPG keys from the distro the migration has
+# been performed from
+# Copyright (c) 2021 EuroLinux
```
[the remark this answers is not in the document]
Comment (tompods) [reply remove_kernels.sh]: Added

File: remove_kernels.sh [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+  # be removed only then - this is explained later on as a comment.
+  latest_eurolinux_kernel_path="$(find /boot -name ${latest_eurolinux_kernel_package/kernel/vmlinuz})"
+  grubby --set-default="${latest_eurolinux_kernel_path}"
+}
+
+update_grub() {
+  # Cover all distros and versions bootloader entries.
+  # This snippet has been copied from migrate2eurolinux.sh and it's about to
+  # be brainstormed what location/responsibility (script) suits it better,
+  # TODO: more EFI entries?
```
[the remark this answers is not in the document]
Comment (tompods) [reply remove_kernels.sh]: Great idea that comes in handy later. Should I also reimplement the same snippet in migrate2eurolinux.sh?

File: remove_kernels.sh (hunk 1 above)
Comment (tompods) [reply remove_kernels.sh:115]: Turns out it wasn't working properly and I simplified it along with fixing the issue.

File: remove_kernels.sh [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+  # /etc/redhat-release and the package "kernel" is EuroLinux-branded
+
+  release_provider="$(rpm -q --whatprovides /etc/redhat-release)"
+  if [[ ! "$release_provider" =~ ^el-release ]]; then
+    exit_message "Could not determine if a migration was successful -
+/etc/redhat-release is provided by the package \"$release_provider\" rather than 
+el-release."
+  fi
+
+  mapfile -t installed_kernel_packages < <(rpm -qa --qf '%{NAME}-%{VERSION}-%{RELEASE}.%{ARCH}|%{VENDOR}|%{PACKAGER}\n' kernel* | sed 's@\ @\_@g')
```
[the remark this answers is not in the document]
Comment (tompods) [reply remove_kernels.sh:46]: Explained

File: remove_kernels.sh [hunk 5]
```diff
@@ -44,28 +60,18 @@ set_latest_eurolinux_kernel() {
...
+  printf "$grub2_conf (symlinked to $(readlink $grub2_conf)).\n"
+  grub2-mkconfig -o "$grub2_conf"
 }
 
 prepare_list_of_kernels_to_be_removed() {
-  # Consider several scenarios such non-EuroLinux kernels that come from e.g.
-  # ELRepo or other sources. What should be done with them is up to the user
-  # to decide via an answer or a parameter.
+  # Consider several scenarios such as non-EuroLinux kernels that come from
+  # e.g.  ELRepo or other sources. What should be done with them is up to the
```
Comment (AlexBaranowski) [inline remove_kernels.sh:73]: double space

**Review by AlexBaranowski — APPROVED** [review]

File: remove_kernels.sh (hunk 5 above)
Comment (tompods) [reply remove_kernels.sh:73]: Fixed in 8d63bdd

**Pull request merged**